src/helper.h

```diff
@@ -60,26 +61,15 @@ class HelperServerBase;
  * If an overloaded helper has been overloaded for 3+ minutes, an attempt to use
  *   it results in on-persistent-overload action, which may kill worker.
  */
-class helper
+class helper: public RefCountable
 {
-    CBDATA_CLASS(helper);
-
 public:
+    using Pointer = RefCount<helper>;
+
+    /// \returns a newly created instance of the named helper
     /// \param name admin-visible helper category (with this process lifetime)
-    inline helper(const char *name) :
-        cmdline(nullptr),
-        id_name(name),
-        ipc_type(0),
-        droppedRequests(0),
-        overloadStart(0),
-        last_queue_warn(0),
-        last_restart(0),
-        timeout(0),
-        retryTimedOut(false),
-        retryBrokenHelper(false),
-        eom('\n') {
-        memset(&stats, 0, sizeof(stats));
-    }
+    static Pointer Make(const char *name);
+
```

Comment:
Please revert the code style to using C++ construction instead of a maker-factory style.

```suggestion
explicit helper(const char *name) : id_name(name) {}
```